pkg/compose/build_buildkit.go

```diff
@@ -47,8 +56,10 @@ func (s *composeService) doBuildBuildkit(ctx context.Context, project *types.Pro
 	defer cancel()
 	w := xprogress.NewPrinter(progressCtx, s.stdout(), os.Stdout, mode)
 
-	// We rely on buildx "docker" builder integrated in docker engine, so don't need a DockerAPI here
-	response, err := build.Build(ctx, driverInfo, opts, nil, filepath.Dir(s.configFile().Filename), w)
+	// Get the DockerAPI if a "docker" export is defined (ie: up and run command), otherwise get nil and let use the default buildx builder
+	dockerAPI := getDockerAPI(s.dockerCli, opts)
+
+	response, err := build.Build(ctx, dis, opts, dockerAPI, filepath.Dir(s.configFile().Filename), w)
```

Comment:
I think it's fine to always pass `dockerAPI` to `build.Build` as it's only used when loading the image in docker store anyway:

```suggestion
	response, err := build.Build(ctx, dis, opts, &internalAPI{s.dockerCli}, filepath.Dir(s.configFile().Filename), w)
```